app/gui/src/controller/searcher.rs

```diff
@@ -1158,6 +1163,21 @@ impl Searcher {
     }
 }
 
+
+// === Searcher helpers ===
+
+fn component_list_builder_with_favorites(
+    groups: Rc<Vec<model::execution_context::ComponentGroup>>,
+    db: Rc<model::SuggestionDatabase>,
+) -> component::builder::List {
+    let mut builder = component::builder::List::new(db);
+    builder.set_favorites(groups.iter());
```

Comment:
The `set_favorites` function should be easier to use. You should be able to write `builder.set_favorites(groups))` instead.

Reply:
You can call `builder.set_favorites(&*groups))` but is it _easier_ is a matter of taste.

On the other hand, having the argument like `impl AsRef<impl IntoIterator<Item = ...>>` seems like an overkill to me.